extensions/wikia/MyHome/SpecialWikiActivity.class.php

```diff
@@ -9,150 +9,183 @@ class SpecialWikiActivity extends UnlistedSpecialPage {
-				}
-				else {
-					$wgOut->wrapWikiMsg( '<div id="myhome-log-in">$1</div>', array('myhome-log-in', wfGetReturntoParam()) );
+			if ( $wgUser->isAnon() ) {
+				if ( get_class( RequestContext::getMain()->getSkin() ) == 'SkinOasis' ) {
+					$wgOut->wrapWikiMsg( '<div class="latest-activity-watchlist-login" >$1</div>', [ 'oasis-activity-watchlist-login', wfGetReturntoParam() ] );
+				} else {
+					$wgOut->wrapWikiMsg( '<div id="myhome-log-in">$1</div>', [ 'myhome-log-in', wfGetReturntoParam() ] );
 				}
 
 				//oasis-activity-watchlist-login
 				// RT #23970
-				$wgOut->addInlineScript(<<<JS
+				$wgOut->addInlineScript( <<<JS
 $(function() {
 	$('#myhome-log-in').find('a').click(function(ev) {
 		openLogin(ev);
 	});
 });
 JS
 				);
-				wfProfileOut(__METHOD__);
+				wfProfileOut( __METHOD__ );
+
 				return;
-			}
-			else {
+			} else {
 				$this->feedSelected = 'watchlist';
 				$feedProxy = new WatchlistFeedAPIProxy();
-				$feedRenderer = new WatchlistFeedRenderer();
 			}
 		} else {
-		//for example: wiki-domain.com/wiki/Special:WikiActivity
+			//for example: wiki-domain.com/wiki/Special:WikiActivity
 			$this->feedSelected = 'activity';
 			$feedProxy = new ActivityFeedAPIProxy();
-			$feedRenderer = new ActivityFeedRenderer();
 		}
 
-		$feedProvider = new DataFeedProvider($feedProxy);
+		$feedProvider = new DataFeedProvider( $feedProxy );
 
 		global $wgJsMimeType, $wgExtensionsPath;
-		$wgOut->addScript("<script type=\"{$wgJsMimeType}\" src=\"{$wgExtensionsPath}/wikia/MyHome/WikiActivity.js\"></script>\n");
-		$wgOut->addExtensionStyle(AssetsManager::getInstance()->getSassCommonURL('extensions/wikia/MyHome/oasis.scss'));
+		$wgOut->addScript( "<script type=\"{$wgJsMimeType}\" src=\"{$wgExtensionsPath}/wikia/MyHome/WikiActivity.js\"></script>\n" );
+		$wgOut->addExtensionStyle( AssetsManager::getInstance()->getSassCommonURL( 'extensions/wikia/MyHome/oasis.scss' ) );
 
-		wfRunHooks( 'SpecialWikiActivityExecute', array( $wgOut, $wgUser ));
+		wfRunHooks( 'SpecialWikiActivityExecute', [ $wgOut, $wgUser ] );
 
-		$data = $feedProvider->get(50);  // this breaks when set to 60...
+		$data = $feedProvider->get( 50 );  // this breaks when set to 60...
 
 		// FIXME: do it in AchievementsII extension
 		global $wgEnableAchievementsInActivityFeed, $wgEnableAchievementsExt;
-		if((!empty($wgEnableAchievementsInActivityFeed)) && (!empty($wgEnableAchievementsExt))){
-			$wgOut->addExtensionStyle("{$wgExtensionsPath}/wikia/AchievementsII/css/achievements_sidebar.css");
+		if ( ( !empty( $wgEnableAchievementsInActivityFeed ) ) && ( !empty( $wgEnableAchievementsExt ) ) ) {
+			$wgOut->addExtensionStyle( "{$wgExtensionsPath}/wikia/AchievementsII/css/achievements_sidebar.css" );
 		}
 
 		// use message from MyHome as special page title
-		$wgOut->setPageTitle(wfMsg('oasis-activity-header'));
+		$wgOut->setPageTitle( wfMsg( 'oasis-activity-header' ) );
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this legacy wfMsg* function call with one of the new i18n functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)